### PR TITLE
feat(dracut.sh): make initramfs-${kernel}.img filename configurable

### LINKED ISSUE #99: detect and generate initrd.img-${kernel}
https://www.freedesktop.org/wiki/Specifications/BootLoaderSpec/ refers to initramfs as "initrd".

One of the most popular distro's (Debian) expects /boot/initrd.img-${kernel} instead of initramfs-${kernel}.img, see https://kernel-team.pages.debian.net/kernel-handbook/ch-packaging.html#s-arch-dep

Let's look for a way to detect in a reliable way what should be the default output file that is more inclusive.

Possibly related openSUSE patch downstream: https://github.com/dracutdevs/dracut/commit/f68649cf55ca4545ed8304435c9c7315846c45a4

### PR DESCRIPTION
**Pull request by LaszloGombos**
## Changes

The ambition is to propose a minimum viable patch and replace simple downstream patches like https://github.com/openSUSE/dracut/commit/1442c9d29

The `initrdname` configuration option is required to match a `initr*${kernel}*` file pattern.

Distributions are incentived but not required to follow the upcoming https://github.com/uapi-group/specifications/issues/100 standard (assuming positive resolution).

## Checklist
- [X] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #99

Tested with the following config file entry

```
initrdname=initrd-${kernel} 
```

### CODE REVIEW

**Review by freswa — APPROVED** [review]
lgtm

File: dracut-initramfs-restore.sh [hunk 1]
```diff
@@ -46,8 +46,15 @@ elif mountpoint -q /efi; then
 elif mountpoint -q /boot/efi; then
     IMG="/boot/efi/$MACHINE_ID/$KERNEL_VERSION/initrd"
 else
-    echo "No initramfs image found to restore!"
-    exit 1
+    files=("/boot/initr*${KERNEL_VERSION}*")
```
Comment (Conan-Kudo) [inline dracut-initramfs-restore.sh:49]: How is this guaranteed to work? Someone could set it to be `$KERNEL_VERSION-initramfs`...

File: man/dracut.conf.5.asc [hunk 2]
```diff
@@ -320,6 +320,12 @@ Logging levels:
    If set to _yes_, try to execute tasks in parallel (currently only supported
    for _--regenerate-all_).
 
+*initrdname=*"_<filepattern>_"::
+    Specifies the file name for the generated initramfs if it is not set otherwise.
+    The initrdname configuration option is required to match the _initr*${kernel}*_
+    file pattern and only one file with this pattern should exists in the
+    directory where initramfs is loaded from.
```
Comment (Conan-Kudo) [inline man/dracut.conf.5.asc:327]: The default name needs to be documented.

File: man/dracut.8.asc [hunk 3]
```diff
@@ -18,13 +18,8 @@ DESCRIPTION
...
 If <kernel version> is omitted, then the version of the actual running
-kernel is used. If <image> is omitted or empty, depending on bootloader
-specification, the default location can be
-_/efi/<machine-id>/<kernel-version>/initrd_,
-_/boot/<machine-id>/<kernel-version>/initrd_,
-_/boot/efi/<machine-id>/<kernel-version>/initrd_,
-_/lib/modules/<kernel-version>/initrd_ or
-_/boot/initramfs-<kernel-version>.img_.
+kernel is used. If <image> is omitted or empty, the default location will be
+determined by the local configuration or Linux distribution policy.
```
Comment (Conan-Kudo) [inline man/dracut.8.asc:22]: This is not actually true. We are only adding customization for the file name, not the paths. So this is a bit excessive.

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: dracut.sh [hunk 4]
```diff
@@ -1093,6 +1093,13 @@ drivers_dir="${drivers_dir%"${drivers_dir##*[!/]}"}"
 [[ $sbat_l ]] && sbat="$sbat_l"
 [[ $machine_id_l ]] && machine_id="$machine_id_l"
 
+if ! [[ $outfile ]] && [[ $initrdname ]]; then
+    outfilename="$initrdname"
```
Comment (aafeijoo-suse) [inline dracut.sh:1097]: `outfilename` is assigned and never used.

File: dracut.sh [hunk 5]
```diff
@@ -1093,6 +1093,13 @@ drivers_dir="${drivers_dir%"${drivers_dir##*[!/]}"}"
...
+else
+    # shellcheck disable=SC2034
+    outfilename="initramfs-${kernel}.img"
```
Comment (aafeijoo-suse) [inline dracut.sh:1100]: `outfilename` is assigned and never used.

File: dracut.sh [hunk 6]
```diff
@@ -1093,6 +1093,13 @@ drivers_dir="${drivers_dir%"${drivers_dir##*[!/]}"}"
 [[ $sbat_l ]] && sbat="$sbat_l"
 [[ $machine_id_l ]] && machine_id="$machine_id_l"
 
+if ! [[ $outfile ]] && [[ $initrdname ]]; then
```
Comment (aafeijoo-suse) [inline dracut.sh:1096]: The condition `if [[ $initrdname ]]` should be moved within the next `if ! [[ $outfile ]]`, i.e., if `outfile` has any value, `initrdname` is useless.

**Review by aafeijoo-suse — CHANGES_REQUESTED** [review]

File: man/dracut.8.asc (hunk 3 above)
Comment (pvalena) [reply man/dracut.8.asc:22]: I agree with the change- documentation is aimed at users; who either use the default, or can override it explicitly.

**Review by pvalena — APPROVED** [review]
LGTM.

**Review by Conan-Kudo — APPROVED** [review]

**Pull request merged**